### ISSUE TITLE
--oaa_subsample reports after-training predictions

### ISSUE THREAD

**Issue by denik** [VowpalWabbit/vowpal_wabbit#1344]
When --oaa_subsample option is enabled,  'average loss' reported by vw is no longer progressive validation loss, but is a training loss. This makes it impossible to rely on 'average loss' for model selection.

**Comment by JohnLangford** [comment]
Hmm.  This obviously need fixing.

-John

**Comment by JohnLangford** [comment]
I'm not sure what to do with this one.  Using --oaa_subsample is equivalent to asking to not have a complete prediction.  The current output _is_ a form of progressive validation, except with the flaw that the subsampled classes need to be upweighted.

Perhaps we should modify this to form an unbiased estimate of the full progressive validation loss?
